src/QorusAuthenticator.ts

```diff
@@ -401,7 +424,7 @@ const _QorusAuthenticator = (): Authenticator => {
   };
 
   /**
-   * A setter to set the url of the selected endpoint
+   * A setter to set the url of the selected {@link Endpoint}
    */
   const setEndpointUrl = async (url: string, id: string = selectedEndpoint.id): Promise<string | undefined> => {
```

Comment:
This is going to crash if this function is called and `selectedEndpoint` is not defined yet. Change it to `= selectedEndpoint?.id`

Reply:
It's strange that TypeScript did no complain about this. Do we have `strictNullChecks` enabled?

Reply:
yes we do have the check in tsconfig